zulip_bots/zulip_bots/bots/giphy/giphy.py

```diff
@@ -75,8 +112,11 @@ def get_url_gif_giphy(keyword: str, api_key: str) -> Union[int, str]:
     data.raise_for_status()
 
     try:
-        gif_url = data.json()['data']['images']['original']['url']
-    except (TypeError, KeyError):  # Usually triggered by no result in Giphy.
+        if count == 1:
```

Comment:
I wasn't reviewing this, was just curious about progress.  This isn't directly related to your fix, but it would be nice if we could clean up this function so it doesn't have all the conditional branching on count-equals-one-vs.-otherwise.  It would be nice to avoid having two different return types.  I know we have to a conditional **somewhere** to handle the only-one and only-zero cases, but if we could organize the code to have that happen in as few places as possible, that would be nice.

Reply:
Okay, I'll clean this function now... this was the result of "hack mode" advise... though it was super hacky I think :sweat_smile: